### PR TITLE
feat(medusa): Expose session options

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

File: packages/medusa/src/types/global.ts [hunk 1]
```diff
@@ -100,10 +100,19 @@ export type ModuleExports = {
...
 }
 
+type SessionOptions = {
+  resave?: boolean
+  saveUninitialized?: boolean
+  secret?: string
+  ttl?: number
+}
+
 export type ConfigModule = {
```
Comment (olivermrbl) [inline packages/medusa/src/types/global.ts:110]: self-review: At some point, we should change the config to follow our casing convention. 

Not super pretty to have `projectConfig > some_config > someOtherConfig`

File: packages/medusa/src/loaders/express.ts [hunk 2]
```diff
@@ -22,17 +22,17 @@ export default async ({ app, configModule }: Options): Promise<Express> => {
     sameSite = "none"
   }
 
-  const { cookie_secret } = configModule.projectConfig
+  const { cookie_secret, session_options } = configModule.projectConfig
   const sessionOpts = {
-    resave: true,
-    saveUninitialized: true,
+    resave: session_options?.resave ?? true,
```
Comment (carlos-r-l-rodrigues) [inline packages/medusa/src/loaders/express.ts:27]: I believe `resave` can be false by default.

Comment (olivermrbl) [reply packages/medusa/src/loaders/express.ts:27]: Added this to not "break" any existing session management, even though it is highly unlikely that any businesses actually use `resave`

**Review by srindom — APPROVED** [review]
LGTM!

**Comment by pevey** [comment]
It would be great if you also exposed the rolling option.  The default is false.

**Comment by olivermrbl** [comment]
@pevey I'll look into it when time allows for it. I am happy to accept a PR for it, if you are up for it :)

**Comment by olivermrbl** [comment]
@pevey That would be great!

Not entirely sure how we ended up with `cookieName` in the first place. `name` is correct according to their docs. And yes, please use `connect.sid` as default to stay backward compatible with users who might rely on that :)

**Pull request merged**